src/Impl/GuzzleHttpRequester.php

```diff
@@ -96,15 +109,23 @@ private function parseFailureResponse(ResponseInterface $response): FailureRespo
     /**
      * @throws StreamxClientException
      */
-    private function parseResponseToJson(ResponseInterface $response)
+    private function parseResponseToJsons(ResponseInterface $response)
     {
-        $jsonString = (string)$response->getBody();
-        $jsonObject = json_decode($jsonString);
-        if (json_last_error() !== JSON_ERROR_NONE) {
-            throw new StreamxClientException(sprintf('Communication error. Response status: %s. Message: %s',
-                $response->getStatusCode(), 'Response could not be parsed.'));
+        $compositeResponseJson = (string)$response->getBody();
+        $singleResponseJsons = explode("\n", $compositeResponseJson); // TODO assuming the only newline characters possible in response, is the one that separates multiple MessageStatus jsons
```

Comment:
It is a wrong assumption.

Reply:
I think wrapping the concatenated single jsons response in parent `{` and `}` - and then parsing root content as list of objects should do it